rotkehlchen/chain/ethereum/modules/juicebox/decoder.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from typing import Any
+
+import requests
+
+from rotkehlchen.chain.ethereum.abi import decode_event_data_abi_str
+from rotkehlchen.chain.ethereum.modules.juicebox.constants import (
+    CPT_JUICEBOX,
+    JUICEBOX_IPFS_GATEWAY,
+    JUICEBOX_PROJECTS,
+    LEGAL_DEFENSE_TORNADO_PROJECT_ID,
+    PAY_ABI,
+    PAY_SIGNATURE,
+    TERMINAL_3_1_2,
+)
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.constants.assets import A_ETH
+from rotkehlchen.db.settings import CachedSettings
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class JuiceboxDecoder(DecoderInterface):
+
+    def _query_project_name(self, project_id: int) -> dict[str, Any] | None:
+        """Query metadata for project id in ipfs"""
+        try:
+            ipfs_hash = self.evm_inquirer.contracts.contract(JUICEBOX_PROJECTS).call(
+                node_inquirer=self.evm_inquirer,
+                method_name='metadataContentOf',
+                arguments=[project_id, 0],  # 0 is the ipfs hash of the data
+            )
+        except RemoteError:
+            log.error(f'Failed to query ipfs hash for juicebox project {project_id}')
+            return None
+
+        metadata_url = JUICEBOX_IPFS_GATEWAY + ipfs_hash
+        try:
+            log.debug(f'Querying juicebox metadata at {metadata_url}')
+            metadata: dict[str, Any] = requests.get(
+                url=metadata_url,
+                timeout=CachedSettings().get_timeout_tuple(),
+            ).json()
+        except requests.RequestException as e:
```

Comment:
you also need to catch json decoding error here. Not sure if you get any additional error which may make sense to conditionally alter the log exeption but probably not?
```suggestion
        except (requests.RequestException, json.JSONDecodeError) as e:
```

Reply:
Commented in github about it 
![imagen](https://github.com/rotki/rotki/assets/5068010/6dd056a5-ba99-4cec-9ac7-e1fcd4efdf09)